api/authorization-api-1_0.md

```diff
@@ -889,107 +889,149 @@ The following is a non-normative example of a response to an Access Evaluations
+# Search APIs {#search}
 
-The Subject Search API is based on the Access Evaluation information model, but omits the Subject ID.
+The Search APIs define the message exchange pattern between a client (PEP) and an authorization service (PDP) for returning all subjects, resources, or actions that match the search criteria.
 
-## Subject Search Semantics
+## Search Semantics {#search-semantics}
 
-While the evaluation of a search is implementation-specific, it is expected that any returned results that are then fed into an `evaluation` call MUST result in a `"decision": true` response.
+While the evaluation of a search is implementation-specific, it is expected that any returned results that are then fed into an Access Evaluation API call MUST result in a `"decision": true` response.
 
-In addition, it is RECOMMENDED that a subject search is performed transitively, traversing intermediate attributes and/or relationships. For example, if the members of group G are designated as viewers on a document D, then a search for all users that are viewers of document D will include all the members of group G.
+In addition, it is RECOMMENDED that a search be performed transitively, traversing intermediate attributes and/or relationships. For example, if user U is a member of group G, and group G is designated as a viewer on a document D, then a search for all subjects that can view document D will include user U.
 
-## The Subject Search API Request {#subject-search-request}
+## Search API Request {#search-request}
 
-The Subject Search request is an object consisting of the following entities:
+A search request is constructed based on the Subject, Resource, Action and Context entities defined in the Information Model ({{information-model}}). For a search request, the unique identifier of the entity being searched for is omitted. Additionally, a `page` object as defined in {{search-request-pagination}} MAY be provided.
 
-`subject`:
-: REQUIRED. The subject (or principal) of type Subject.  NOTE that the Subject type is REQUIRED but the Subject ID can be omitted, and if present, is IGNORED.
+The following is a non-normative example of a request to retrieve the resources that user `alice@example.com` can read:
 
-`action`:
-: REQUIRED. The action (or verb) of type Action.
+~~~ json
+{
+  "subject": {
+    "type": "user",
+    "id": "alice@example.com"
+  },
+  "action": {
+    "name": "can_read"
+  },
+  "resource": {
+    "type": "account"
+  }
+}
+~~~
+{: #search-request-example title="Example Search Request"}
 
-`resource`:
-: REQUIRED. The resource of type Resource.
+### Pagination Request {#search-request-pagination}
 
-`context`:
-: OPTIONAL. Contextual data about the request.
+A search request MAY include a `page` object to control pagination. This object can contain the following keys:
 
-`page`:
-: OPTIONAL. A page token for paged requests.
+`next_token`: 
```

Comment:
This is the value retrieved from `next_token` in the previous response. Should this key then not be named `token`?